### PR TITLE
chore: Update ruby and docker base image to 3.0.4

### LINKED ISSUE #4658: fix security issues in docker images
**Describe the bug**

- fix security issues in docker images reported from synk scans

### PR DESCRIPTION
**Pull request by vishnu-narayanan**
# Pull Request Template

## Description

- Update ruby to the latest stable patch version `3.0.4`
- Update the docker base image to resolve security issues reported by `synk`

Fixes #4658 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] This change requires a documentation update

## How Has This Been Tested?

- [x] tests on circle ci
- [x] CE edition tests via gh action
- [ ] docker image builds via docker hub CI

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by vishnu-narayanan** [comment]
@sojan-official Updating `ruby `to `3.1.2`is causing issues with `google-protobuf` package.

**Comment by pranavrajs** [comment]
@vishnu-narayanan Why don't take it in steps? Upgrade to 3.0.4 so that the security issue is fixed and 3.1.2 later.

**Comment by vishnu-narayanan** [comment]
@pranavrajs `3.0.4` was failing with https://hub.docker.com/repository/registry-1.docker.io/chatwoot/chatwoot/builds/e03d7dfd-2a3a-4c75-a91c-3302c22dc131. 

So 
1. Wanted to verify if this error existed in `3.1.2`.
2. Since we are spending the effort to upgrade, it would have been efficient to do it once. 😓 No luck here though.

cc: @sojan-official

**Comment by sojan-official** [comment]
@vishnu-narayanan  should we try upgrading node? 
https://github.com/FrontendMasters/fullstack-graphql/issues/9

File: deployment/chatwoot-worker.1.service [hunk 1]
```diff
@@ -16,10 +16,10 @@ KillMode=mixed
 StandardInput=null
 SyslogIdentifier=%p
 
-Environment="PATH=/home/chatwoot/.rvm/gems/ruby-3.0.2/bin:/home/chatwoot/.rvm/gems/ruby-3.0.2@global/bin:/home/chatwoot/.rvm/rubies/ruby-3.0.2/bin:/home/chatwoot/.rvm/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin:/home/chatwoot/.rvm/bin:/home/chatwoot/.rvm/bin"
```
Comment (pranavrajs) [inline deployment/chatwoot-worker.1.service:19]: @vishnu-narayanan Should we wait for release for this change?

Comment (vishnu-narayanan) [reply deployment/chatwoot-worker.1.service:19]: Even if someone is running our `develop in prod, should be fine.  We replace the `systemd` service files as part of every upgrade.

**Pull request merged**